handler/file/range_reader.go

```diff
@@ -0,0 +1,64 @@
+package file
+
+import (
+	"errors"
+	"io"
+)
+
+// rangeReader reads data from one individually retrievable file range.
+type rangeReader struct {
+	// offset is the absolute offset within file where the next read will get
+	// data from.
+	offset    int64
+	reader    io.ReadCloser
+	remaining int64
+}
+
+func (rr *rangeReader) inRange(pos int64) bool {
```

Comment:
Probably could use a different name to make it clear that this also involves current offset, not original range, `inRemainingRange`? I stumbled on this in my first read-through.